core/src/main/java/org/radargun/traits/Streamable.java

```diff
@@ -0,0 +1,24 @@
+package org.radargun.traits;
+
+import org.radargun.Operation;
+
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * @author Roman Macor &lt;rmacor@redhat.com&gt;
+ */
+@Trait(doc = "Provides interface for executing distributed tasks on streams")
+public interface Streamable {
+   String TRAIT = Streamable.class.getSimpleName();
+   Operation STREAMABLE = Operation.register(TRAIT + ".Streamable");
+
+   Stream stream(String resourceName);
+
+   Stream parallelStream(String resourceName);
+
+   interface StreamConsumer extends Consumer<Stream> {
```

Comment:
If you're expecting a result, it's not just a consumer, it's https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html